types/datum.go

```diff
@@ -1122,7 +1122,7 @@ func (d *Datum) convertToMysqlTimestamp(sc *stmtctx.StatementContext, target *Fi
 		t   Time
 		err error
 	)
-	fsp := DefaultFsp
+	fsp := MaxFsp
```

Comment:
this change will cause some automated-tests fail, be cautious with that since no CI test for now